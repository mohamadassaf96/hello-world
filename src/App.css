.App {
  text-align: center;
}

body {
  background-color: lightblue;
}

div{
  margin-top: 100px;
  text-align: center;
}

.search-form 
{
  padding-top: 20px;
}

input 
{
  margin-left: 20px;
}

button
{
  margin-left: 20px;
}

.container ul {
  margin-top: 50px;
  list-style-type: none;
}
.container ul li {
  height: 30px;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
