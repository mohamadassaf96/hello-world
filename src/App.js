import React, { Suspense } from 'react';
import ShoppingList from "./ShoppingList.js"
import Tour from 'reactour';

const steps = [
  {
    selector: '.first-step',
    content: 'Welcome to Shopping List Manager!',
  },
  {
    selector: '[data-tut="reactour_item"]',
    content: 'Type the name of the item here',
    position: 'right',
  },
  {
    selector: '[data-tut="reactour_submit"]',
    content: 'Press this button to add to the list',
    position: 'right',
  },
  {
    selector: '[data-tut="reactour_search"]',
    content: 'Use this form to search to filter list items',
    position: 'right',
  },
]

class App extends React.Component 
{
  constructor(props) {
    super(props);
    this.state = { items: [], filteredItems: [], text: '', search: '', isTourOpen: false};
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.updateSearch = this.updateSearch.bind(this);
  }

  setOpen = (b) => 
  {
    this.setState({isTourOpen: b});
  }

  render() {
    return (
      <div className="container">
        <div>
          <button onClick={this.handleClick} style={{margin: 20, fontSize: 15}}>
            Start tutorial
          </button>
        </div>
        <h3 style={{textAlign: "center"}}>Shopping List</h3>
        <form style={{textAlign: "center"}} onSubmit={this.handleSubmit}>
          <label htmlFor="new_item">
            What do you need to buy?
          </label>
          <input data-tut="reactour_item" style={{marginLeft: 10}}
            id="new_item"
            onChange={this.handleChange}
            value={this.state.text}
          />
          <button data-tut="reactour_submit">
            Add #{this.state.items.length + 1}
          </button>
        </form>
        <form 
          style={{textAlign: "center", marginTop: 20}} 
          className="search-form"
        >
          <label htmlFor="search_item">
            Search for
          </label>
          <input
          id="search_item"
          value={this.state.search}
          onChange={this.updateSearch}
          style={{marginLeft: 10}}
          data-tut="reactour_search" 
          />
        </form>
        <ShoppingList items={this.state.filteredItems} />
      <Suspense fallback={<React.Fragment />}>
        <Tour
          steps={steps}
          isOpen={this.state.isTourOpen}
          onRequestClose={() => this.setOpen(false)}
          />
      </Suspense>
      </div>
    );
  }

  handleChange(e) 
  {
    this.setState({ text: e.target.value });
  }

  handleClick = () => 
  {
    this.setOpen(true);
  }

  updateSearch(e) 
  {
    let tempFilteredItems = e.target.value.length === 0 ? this.state.items : this.state.items.filter(
      (item) => {
        return item.text.indexOf(e.target.value) !== -1;
      }  
    );
    this.setState({ search: e.target.value });
    this.setState({ filteredItems: tempFilteredItems});
  }

  handleSubmit(e) 
  {
    e.preventDefault();
    if (this.state.text.length === 0) {
      return;
    }
    const newItem = {
      text: this.state.text,
      id: Date.now()
    };
    this.setState(state => ({
      items: state.items.concat(newItem),
      filteredItems: state.items.concat(newItem),
      text: ''
    }));
  }

}

export default App;