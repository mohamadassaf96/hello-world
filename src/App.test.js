import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import App from './App';
import ReactTestUtils from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';
import Enzyme from 'enzyme';
import Simulate from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });
let wrapper = null;

describe('App Component', function () {

  beforeEach(() => {
    wrapper = mount(
            <App />
        );
  });

  test('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Input works", () => {
    wrapper.find("input").at(0).simulate("change", { target: {value: "oranges"} });
    wrapper.find("button").at(1).simulate("submit");
    (expect(wrapper.find("ShoppingList").at(0).contains("oranges"))).toBeTruthy();
  });

  test("Empty input works", () => {
    wrapper.find("input").at(0).simulate("change", { target: {value: ""} });
    wrapper.find("button").at(1).simulate("submit");
    expect(wrapper.state().items.length).toEqual(0);
  });

  test("search works", () => {
    wrapper.find("input").at(0).simulate("change", { target: {value: "oranges"} } );
    wrapper.find("button").at(1).simulate("submit");
    wrapper.find("input").at(0).simulate("change", { target: {value: "apples"} } );
    wrapper.find("button").at(1).simulate("submit");
    wrapper.find("input").at(0).simulate("change", { target: {value: "watermelon"} } );
    wrapper.find("button").at(1).simulate("submit");
    expect(wrapper.find("ShoppingList").contains("oranges")).toBeTruthy();
    expect(wrapper.find("ShoppingList").contains("apples")).toBeTruthy();
    expect(wrapper.state().items.length).toEqual(3);
    wrapper.find("input").at(1).simulate("change", { target: {value: "or"} } );
    expect(wrapper.find("ShoppingList").contains("oranges")).toBeTruthy();
    expect(wrapper.state().filteredItems.length).toEqual(1);
  });

  test("Empty search works", () => {
    wrapper.find("input").at(0).simulate("change", { target: {value: "oranges"} } );
    wrapper.find("button").at(1).simulate("submit");
    wrapper.find("input").at(0).simulate("change", { target: {value: "apples"} } );
    wrapper.find("button").at(1).simulate("submit");
    wrapper.find("input").at(0).simulate("change", { target: {value: "watermelon"} } );
    wrapper.find("button").at(1).simulate("submit");
    wrapper.find("input").at(1).simulate("change", { target: {value: ""} } );
    expect(wrapper.state().filteredItems.length).toEqual(3);
  });
});

