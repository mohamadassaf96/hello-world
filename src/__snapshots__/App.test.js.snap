// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Renders correctly 1`] = `ReactWrapper {}`;

exports[`SMSNav Component Renders correctly 1`] = `null`;
